brightlayer-ui/react-component-library
# Feature/2704 migrate v4 to v5

Fixes 2704 .


#### Changes proposed in this Pull Request:
- [X] Migrate MUI v4 to V5
- [X] Fix import statements
- [ ] Test cases need to fix (mui v5 has removed `/core/test-utils` https://stackoverflow.com/questions/67307897/react-material-ui-v5-createshallow-testing-util-unable-to-be-imported) - This will be part of different issue [378](https://github.com/brightlayer-ui/react-component-library/issues/378)
- [x] Need to fix InfolistItem ripple feature (button prop has been removed from ListItemProps).

#### Screenshots:
<img width="1776" alt="Screenshot 2022-02-15 at 7 18 18 PM" src="https://user-images.githubusercontent.com/10433274/154075092-2b297c46-6753-4bb5-ae8a-39c7695740ab.png">
<img width="1698" alt="Screenshot 2022-02-15 at 7 18 01 PM" src="https://user-images.githubusercontent.com/10433274/154075119-fd546d88-d3c9-4b96-9608-295bc44b5110.png">


#### To Test:
- For testing use showcase demo branch https://github.com/brightlayer-ui/react-showcase-demo/tree/feature/2711-upgrate-mui-to-v5 (it's in progress)
- For mui-icons use https://github.com/brightlayer-ui/icons/pull/188.

**Note: React themes repo still need to update. So, if you see error related to themes please ignore for now .**
As per discussion we are going to raise PR agains mui-v5 branch. Once every issue has been resolved will merge into dev.
